Adds rules to build a shared library version of talloc.

diff -ur talloc-1.0/Makefile.in talloc-1.0-debian/Makefile.in
--- talloc-1.0/Makefile.in	2007-04-25 16:18:22.000000000 +0200
+++ talloc-1.0-debian/Makefile.in	2007-04-30 02:26:26.000000000 +0200
@@ -15,11 +15,17 @@
 CFLAGS = @CFLAGS@ -DHAVE_CONFIG_H= -I. -I@srcdir@ -I@libreplacedir@
 EXTRA_TARGETS = @DOC_TARGET@
 
-.SUFFIXES: .c .o .3 .3.xml .xml .html
+.SUFFIXES: .c .o .po .3 .3.xml .xml .html
 
 LIBOBJ = @TALLOCOBJ@ @LIBREPLACEOBJ@
 
-all: showflags libtalloc.a testsuite $(EXTRA_TARGETS)
+SOLIB = libtalloc.so.1.0.0
+SONAME = libtalloc.so.1
+
+.c.po:
+	$(CC) -fPIC -o $@ -c $< $(CFLAGS)
+
+all: showflags libtalloc.a $(SOLIB) testsuite $(EXTRA_TARGETS)
 
 showflags:
 	@echo 'talloc will be compiled with flags:'
@@ -33,10 +39,14 @@
 	ar -rv $@ $(LIBOBJ)
 	@-ranlib $@
 
+$(SOLIB): $(LIBOBJ:.o=.po)
+	$(CC) -Wl,-soname,$(SONAME) -shared -o $@ $^
+
 install: all 
 	${INSTALLCMD} -d $(DESTDIR)${libdir}
 	${INSTALLCMD} -d $(DESTDIR)${libdir}/pkgconfig
 	${INSTALLCMD} -m 755 libtalloc.a $(DESTDIR)$(libdir)
+	${INSTALLCMD} -m 755 $(SOLIB) $(DESTDIR)$(libdir)
 	${INSTALLCMD} -d $(DESTDIR)${includedir}
 	${INSTALLCMD} -m 644 $(srcdir)/talloc.h $(DESTDIR)$(includedir)
 	${INSTALLCMD} -m 644 talloc.pc $(DESTDIR)$(libdir)/pkgconfig
@@ -52,7 +62,7 @@
 	-test -z "$(XSLTPROC)" || $(XSLTPROC) --nonet -o $@ http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl $<
 
 clean:
-	rm -f *~ $(LIBOBJ) libtalloc.a testsuite testsuite.o *.gc?? talloc.3 talloc.3.html
+	rm -f *~ $(LIBOBJ) $(SOLIB) libtalloc.a testsuite testsuite.o *.gc?? talloc.3 talloc.3.html
 
 test: testsuite
 	./testsuite
